src/core/markovProcessGraph.js

```diff
@@ -679,21 +710,24 @@ export class MarkovProcessGraph {
   }
 
   toMarkovChain(): OrderedSparseMarkovChain {
-    const nodeOrder = Array.from(this._nodes.keys()).sort();
+    // Array-ify the iterators so we can iterate over them multiple times
+    // without needing to re-generate them.
```

Comment:
No need: just iterate over `this._nodes`/`this._edges` as much as you
need. You’re inside the abstraction boundary.